src/verification/verification.service.ts

```diff
@@ -0,0 +1,37 @@
+import { JwtService } from '@nestjs/jwt';
+import { User } from '@prisma/client';
+import { VerificationToken } from '../common/interfaces/verification-token.interface';
+import { UserService } from '../user/user.service';
+import {
+  BadRequestException,
+  Inject,
+  Injectable,
+  forwardRef,
+} from '@nestjs/common';
+
+@Injectable()
+export class VerificationService {
+  constructor(
+    private readonly jwtService: JwtService,
+    @Inject(forwardRef(() => UserService))
+    private readonly userService: UserService,
+  ) {}
+
+  async generateVerificationToken(user: User): Promise<VerificationToken> {
+    const payload = { email: user.email, sub: user.id };
+    return {
+      verificationToken: this.jwtService.sign(payload),
+    };
+  }
+
+  async verifyAccount(verificationToken: string): Promise<void> {
+    try {
+      await this.jwtService.verify(verificationToken);
+      const userToBeVerified =
+        await this.userService.finsOneByVerificationToken(verificationToken);
+      await this.userService.updateUserVerification(userToBeVerified.id);
```

Comment:
If you create a module that is (or should be) responsible for verifying tokens. Updating user should not be a part of it's responsibility. 
It forces you to create a circular dependency between UserModule and VerificationModule